napari/utils/events/evented_model.py

```diff
@@ -192,26 +193,49 @@ class Config:
         # NOTE: json_encoders are also added EventedMetaclass.__new__ if the
         # field declares a _json_encode method.
         json_encoders = _BASE_JSON_ENCODERS
+        # this custom use of allow mutation behaves as normal (0 is false, 1 and 2 are true)
+        # but allows us to use 2 as a special value meaning "inplace mutation"
+        allow_mutation = 2
 
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
 
         self._events.source = self
         # add event emitters for each field which is mutable
-        event_names = [
+        field_events = [
             name
             for name, field in self.__fields__.items()
             if field.field_info.allow_mutation
         ]
-        event_names.extend(self.__property_setters__)
-        self._events.add(**dict.fromkeys(event_names))
+
+        self._events.add(
+            **dict.fromkeys(field_events + list(self.__property_setters__))
+        )
+
+        # hook up events from children
+        for name in field_events:
+            value = getattr(self, name)
+            if hasattr(value, 'events'):
+                # TODO: won't track source all the way in?
```

Comment:
I think this is still relevant right? If so, consider writing a test to clarify what the desired/expected behavior is.

Reply:
I *think* it's actually fine: if you change `model.x`, the source of `model.events.x` will be `model`, while the source of `model.x.events` will be `x`. But I'll make some tests for it.